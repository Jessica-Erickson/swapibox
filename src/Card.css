.Card {
  box-shadow: 0 0 0 0 black;
  height: 250px;
  width: 100px;
  transition-property: box-shadow;
  transition-delay: 2s;
  transition-duration: 2s;
}

.saber {
  background-color: transparent;
  box-shadow: 0 0 0 0 transparent;
  height: 5px;
  position: absolute;
  left: 10px;
  width: 0px;
  transition-property: width;
  transition-duration: 2s;
}

.isLit {
  box-shadow: 0 0 5px -2px lightblue, 0 0 5px -2px white;
  width: 80px;
}

.isAlsoLit {
  box-shadow: 0 0 50px -15px lightblue, 0 0 50px -15px white;
}

img {
  transform: rotate(225deg);
}