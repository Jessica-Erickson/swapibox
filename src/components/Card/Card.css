.Card {
  background-color: #444;
  border-radius: 25px;
  box-shadow: 0 0 0 0 transparent;
  display: grid;
  font-weight: 700;
  grid-template: 'heading hilt' 50px
                 'content content' 1fr
                  / 1fr 90px;
  letter-spacing: 1px;
  margin: 25px;
  width: 300px;
  z-index: 0;
}

.saber {
  margin-top: 27px;
  border-radius: 5px;
  grid-area: heading;
  height: 10px;
  justify-self: end;
  z-index: 3;
}

.hilt {
  grid-area: hilt;
  margin-left: 1px;
  transform: rotate(225deg);
  z-index: 6;
}

.hilt:hover {
  cursor: pointer;
}

.name {
  grid-area: heading;
  margin: 15px 0px 0px 20px;
  text-align: center;
  z-index: 9;
}

.content {
  grid-area: content;
  color: white;
  list-style: none;
  margin: 20px 50px 25px 50px;
  padding: 0;
}

.isLit {
  background-color: hsl(195, 85%, 70%);
  box-shadow: 0 0 10px deepskyblue, 0 0 15px white;
  transition: width 1s;
  width: 185px;
}

.isOff {
  background-color: transparent;
  box-shadow: 0 0 0  transparent;
  transition: background-color 0s 1.8s , box-shadow 0s 1.8s, width 1s 0.8s;
  width: 1px;
}

.isAlsoLit {
  box-shadow: 0 0 50px -5px deepskyblue, 0 0 60px -5px white;
  transition: box-shadow 1s 0.8s;
}

.isAlsoOff {
  box-shadow: 0 0 0 0 transparent;
  transition: box-shadow 1s;
}

.isBlack {
  color: black;
  transition: color 1s;
}

.isWhite {
  color: white;
  transition: color 1s 0.8s;
}