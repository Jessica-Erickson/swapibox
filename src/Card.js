import React, { Component } from 'react';
import './Card.css';
const src1 = require('./lightsaber-wt.png');
const src2 = require('./lightsaber-bk.png');

export default class Card extends Component {
  constructor() {
    super();

    this.state = {
      isActive: false,
      src: src1
    }
  }

  handleSaberClick = () => {
    this.setState({ isActive: !this.state.isActive });
  }

  render() {
    return (
      <article className={'Card ' + (this.state.isActive ? 'isAlsoLit' : 'isAlsoOff')}>
        <h3 className={'name ' + (this.state.isActive ? 'isBlack' : 'isWhite')}>Luke Skywalker</h3>
        <div className={'saber ' + (this.state.isActive ? 'isLit' : 'isOff')} ></div>
        <img
          className='hilt'
          src={this.state.src}
          onMouseOver={() => {this.setState({ src: src2 })}}
          onMouseOut={() => {this.setState({ src: src1 })}}
          alt='The hilt of a lightsaber. Turn the lightsaber on to favorite this card.'
          onClick={() => {
            this.setState({ src: src1 });
            this.handleSaberClick();
          }} />
      </article>
    )
  }
}