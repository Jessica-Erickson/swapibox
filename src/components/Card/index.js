import React, { Component } from 'react';
import PropTypes from 'prop-types';
import whiteSaber from './../../assets/icons/lightsaber-wt.png';
import blackSaber from './../../assets/icons/lightsaber-bk.png';
import './Card.css';

class Card extends Component {
  constructor () {
    super();
    this.state = {
      isActive: false,
      src: whiteSaber
    }
  }

  handleSaberClick = () => {
    this.setState({ isActive: !this.state.isActive });
  }

  createListItems = (contentsList) => {
    const name = contentsList.shift()
    return contentsList.map(item => <li key={name[1]}>{item.join(': ')}</li>)
  }

  render() {
    const { isActive , src } = this.state;
    const { contents } = this.props;
    const contentsList = Object.entries(contents)

    return (
      <article className={'Card ' + (isActive ? 'isAlsoLit' : 'isAlsoOff')}>
        <h3 className={'name ' + (isActive ? 'isBlack' : 'isWhite')}>
          {contents.name}
        </h3>
        <div className={'saber ' + (isActive ? 'isLit' : 'isOff')} ></div>
        <img
          className='hilt'
          src={src}
          onMouseOver={() => {this.setState({ src: blackSaber })}}
          onMouseOut={() => {this.setState({ src: whiteSaber })}}
          alt='The hilt of a lightsaber. Turn the lightsaber on to favorite this card.'
          onClick={() => {
            this.setState({ src: whiteSaber });
            this.handleSaberClick();
          }} />
        <ul className='content'>
          {this.createListItems(contentsList)}
        </ul>
      </article>
    )
  }
}

Card.propTypes = {
  contents: PropTypes.object.isRequired
}

export default Card;
