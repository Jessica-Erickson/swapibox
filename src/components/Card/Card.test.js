import React from 'react'
import { shallow } from 'enzyme'
import Card from './index'

describe('Card', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Card />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    wrapper = shallow(<Card />, { disableLifecycleMethods: true });

    expect(wrapper.state()).toMatchSnapshot();
  });

  it('should become active when clicked', () => {
    wrapper = shallow(<Card />);

    expect(wrapper.state('isActive')).toEqual(false);

    wrapper.find('.hilt').simulate('click');

    expect(wrapper.state('isActive')).toEqual(true);
  })
});
