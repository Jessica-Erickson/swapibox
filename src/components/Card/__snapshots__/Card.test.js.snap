// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should have default state 1`] = `
Object {
  "isActive": false,
  "src": "lightsaber-wt.png",
}
`;

exports[`Card should match the snapshot 1`] = `
<article
  className="Card isAlsoOff"
>
  <h3
    className="name isWhite"
  >
    Luke Skywalker
  </h3>
  <div
    className="saber isOff"
  />
  <img
    alt="The hilt of a lightsaber. Turn the lightsaber on to favorite this card."
    className="hilt"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    src="lightsaber-wt.png"
  />
  <ul
    className="content"
  >
    <li>
      Homeworld: 
      Tatooine
    </li>
    <li>
      Species: 
      Human
    </li>
    <li>
      Population: 
      200000
    </li>
  </ul>
</article>
`;
